### ISSUE TITLE
Terraform sees a diff on the scheduling block that won't go away.

### ISSUE THREAD

**Issue by imerchant-clgx** [terraform-providers/terraform-provider-google#4667]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Terraform Version
Terraform v0.11.13

### Affected Resource(s)

* google sole tenant node VM

### Terraform Configuration Files

```tf
scheduling { 
automatic_restart = true 
on_host_maintenance = "MIGRATE" 
preemptible = false 
node_affinities = "${var.st_node_affinity}" 
} 
```

### Expected Behavior

While applying the terraform code, objects which have not changed shouldn't be updated. Here the terraform code thinks the scheduling block has been updated. Which is a bug.

### Actual Behavior

The modification fails with error message:
Error: Error applying plan:

1 error(s) occurred:

module.instance.google_compute_instance.vm_instance: 1 error(s) occurred:

google_compute_instance.vm_instance: Error updating scheduling policy: googleapi: Error 400: Cannot update scheduling for instance with node affinities., badRequest

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with any resources that successfully completed. Please address the error above and apply again to incrementally change your infrastructure.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

I had opened an issue with Hashicorp (22693) which was sent to google provider (4411). But this has come back as a bug with terraform. Can we please request this issue to be escalated. This is causing a lot of delay in our projects.
NOTE: Please do not bump this issue to google as it is not a google issue.

### References

https://github.com/hashicorp/terraform/issues/22693
https://github.com/terraform-providers/terraform-provider-google/issues/4411

* #0000

**Comment by slevenick** [comment]
Hey @imerchant-clgx, sorry this isn't working for you.

Could you provide a more complete config, with the diff that you are seeing? Including the debug logs would help me find the issue happening here, could you include those as well?

And what version of the terraform provider are you using? I am able to create an instance with a scheduling block including node affinities that does not cause a diff, so maybe it is specific to your version?

**Comment by imerchant-clgx** [comment]
Hello @slevenick. I have attached all the config files here. However they are encrypted. Please let me know on what email can I send you the password. This folder has all the config files as well as the crash.log
[GCPCase.zip](https://github.com/terraform-providers/terraform-provider-google/files/3731532/GCPCase.zip)

**Comment by slevenick** [comment]
@imerchant-clgx my username @ google.com

**Comment by slevenick** [comment]
Closing as a duplicate of https://github.com/terraform-providers/terraform-provider-google/issues/4411

This is blocked on the actual issue within the terraform-sdk-plugin here: https://github.com/hashicorp/terraform-plugin-sdk/issues/98
